### PR TITLE
Navigate to /cases after submitting the CaseForm

### PR DESCRIPTION
**Pull request by allysonjp715**
- After hitting submit the CaseForm is closed and the LinelistTable data is refreshed
 - Removes the add button on the LinelistTable
 - Shows the added/edited case row as highlighted in the LinelistTable
 - Shows success message above LinelistTable

Before adding a row:
Screen Shot 2020-07-16 at 3 54 35 PM

After adding a row:
Screen Shot 2020-07-16 at 3 53 47 PM

### CODE REVIEW

**Comment by allysonjp715** [comment]
[Bug fixes](https://trello.com/c/DKJztzhL/276-bug-fixes)

File: verification/curator-service/ui/src/components/LinelistTable.tsx [hunk 1]
```diff
@@ -67,15 +67,18 @@ interface TableRow {
     admittedToHospital: string;
 }
 
-interface Props {
+interface LocationState {
+    newCaseIds: [string];
+    editedCaseIds: [string];
+}
+
+interface Props extends RouteComponentProps<any, any, LocationState> {
```
Comment (axmb) [inline verification/curator-service/ui/src/components/LinelistTable.tsx:75]: Do you get a warning in VSCode for these `any`s? What about if you run `npm run lint`?

File: verification/curator-service/ui/src/components/LinelistTable.tsx [hunk 2]
```diff
@@ -67,15 +67,18 @@ interface TableRow {
     admittedToHospital: string;
 }
 
-interface Props {
+interface LocationState {
+    newCaseIds: [string];
```
Comment (axmb) [inline verification/curator-service/ui/src/components/LinelistTable.tsx:71]: Intentional `[tuple]` in lieu of `array[]`?

File: verification/curator-service/ui/src/components/LinelistTable.tsx [hunk 3]
```diff
@@ -89,6 +92,15 @@ export default class LinelistTable extends React.Component<
             pageSize: 50,
         };
         this.closeCaseForm = this.closeCaseForm.bind(this);
+        // history.location.state can be updated with newCaseIds, on which we
+        // must refresh the table
+        this.unlisten = this.props.history.listen((_, __) =>
+            this.tableRef.current?.onQueryChange(),
+        );
```
Comment (axmb) [inline verification/curator-service/ui/src/components/LinelistTable.tsx:99]: Neat solution!

File: verification/curator-service/ui/src/components/LinelistTable.tsx (hunk 1 above)
Comment (allysonjp715) [reply verification/curator-service/ui/src/components/LinelistTable.tsx:75]: Yes, I thought we were okay with ignoring warnings for small things like that. But in quick fixes I saw an option that said "Use 'never' instead, this is useful when instantiating generic type parameters that you don't need to know the type of." so switched to that instead.

File: verification/curator-service/ui/src/components/LinelistTable.tsx (hunk 2 above)
Comment (allysonjp715) [reply verification/curator-service/ui/src/components/LinelistTable.tsx:71]: Nope, thanks, changed.

**Pull request merged**